hi-ml/src/health_ml/deep_learning_config.py

```diff
@@ -0,0 +1,722 @@
+
+    @property
+    def perform_cross_validation(self) -> bool:
+        """
+        True if cross validation will be be performed as part of the training procedure.
+
+        :return:
+        """
+        return self.number_of_cross_validation_splits > 1
+
+    def get_effective_random_seed(self) -> int:
+        """
+        Returns the random seed set as part of this configuration. If the configuration corresponds
+        to a cross validation split, then the cross validation fold index will be added to the
+        set random seed in order to return the effective random seed.
+
+        :return:
+        """
+        seed = self.random_seed
+        if self.perform_cross_validation:
+            # offset the random seed based on the cross validation split index so each
+            # fold has a different initial random state.
+            seed += self.cross_validation_split_index
+        return seed
+
+
+class DatasetParams(param.Parameterized):
+    azure_dataset_id: str = param.String(doc="If provided, the ID of the dataset to use when running in AzureML. "
+                                             "This dataset must exist as a folder of the same name in the 'datasets' "
+                                             "container in the datasets storage account. This dataset will be mounted "
+                                             "and made available at the 'local_dataset' path when running in AzureML.")
+    local_dataset: Optional[Path] = \
+        param.ClassSelector(class_=Path, default=None, allow_None=True,
+                            doc="The path of the dataset to use, when training is running outside Azure.")
+    extra_azure_dataset_ids: List[str] = \
+        param.List(default=[], allow_None=False,
+                   doc="This can be used to feed in additional datasets to your custom datamodules. These will be"
+                       "mounted and made available as a list of paths in 'extra_local_datasets' when running in AML.")
+    extra_local_dataset_paths: List[Optional[Path]] = \
+        param.List(class_=Path, default=[], allow_None=False,
+                   doc="This can be used to feed in additional datasets "
+                       "to your custom datamodules when running outside of Azure AML.")
+    dataset_mountpoint: str = param.String(doc="The path at which the AzureML dataset should be made available via "
+                                               "mounting or downloading. This only affects jobs running in AzureML."
+                                               "If empty, use a random mount/download point.")
+    extra_dataset_mountpoints: List[str] = \
+        param.List(default=[], allow_None=False,
+                   doc="The mounting points for the datasets given in extra_azure_dataset_ids, when running in "
+                       "AzureML. Use an empty string for all datasets where a randomly chosen mount/download point "
+                       "should be used.")
```

Comment:
now would be a good time to get rid of this extra... madness. Azure_dataset could be a string or a list of strings to be more pythonic.